File: Java/Eta/ValueAdd/src/main/java/com/refinitiv/eta/valueadd/reactor/RDMDirectoryMsgEvent.java
/*|-----------------------------------------------------------------------------
 *|            This source code is provided under the Apache 2.0 license
 *|  and is provided AS IS with no warranty or guarantee of fit for purpose.
 *|                See the project's LICENSE.md for details.
 *|           Copyright (C) 2019-2022 LSEG. All rights reserved.     
 *|-----------------------------------------------------------------------------
 */

package com.refinitiv.eta.valueadd.reactor;

import com.refinitiv.eta.valueadd.domainrep.rdm.directory.DirectoryMsg;

/**
 * Event provided to RDMDirectoryMsgCallback methods.
 * 
 * @see ReactorMsgEvent
 */
public class RDMDirectoryMsgEvent extends ReactorMsgEvent
{
    DirectoryMsg _directoryMsg;

    RDMDirectoryMsgEvent()
    {
        super();
    }

    void rdmDirectoryMsg(DirectoryMsg rdmDirectoryMsg)
    {
        _directoryMsg = rdmDirectoryMsg;
    }

    /**
     * The DirectoryMsg associated with this message event.
     * 
     * @return DirectoryMsg
     */
    public DirectoryMsg rdmDirectoryMsg()
    {
        return _directoryMsg;
    }
}

File: CSharp/Eta/Src/ValueAdd/Reactor/RDMDirectoryMsgEvent.cs
/*|-----------------------------------------------------------------------------
 *|            This source code is provided under the Apache 2.0 license      --
 *|  and is provided AS IS with no warranty or guarantee of fit for purpose.  --
 *|                See the project's LICENSE.md for details.                  --
 *|           Copyright (C) 2022-2023 Refinitiv. All rights reserved.         --
 *|-----------------------------------------------------------------------------
 */

using LSEG.Eta.ValueAdd.Rdm;

namespace LSEG.Eta.ValueAdd.Reactor
{
    /// <summary>
    /// Event provided to <see cref="IDirectoryMsgCallback"/> methods.
    /// </summary>
    public class RDMDirectoryMsgEvent : ReactorMsgEvent
    {
        /// <summary>
        /// Gets DirectoryMsg associated with this message event.
        /// </summary>
        public DirectoryMsg? DirectoryMsg { get; internal set; }

        /// <summary>
        /// Clears to default values.
        /// </summary>
        public override void Clear()
        {
            base.Clear();
            DirectoryMsg = null;
        }

        /// <summary>
        /// Returns the object back to the pool.
        /// </summary>
        public override void ReturnToPool()
        {
            DirectoryMsg = null;
            base.ReturnToPool();
        }
    }
}
